src/main/java/picard/cmdline/CommandLineProgram.java

```diff
@@ -497,4 +498,18 @@ public static boolean hasWebDocumentation(final Class<?> clazz){
     public static String getFaqLink() {
         return "To get help, see http://broadinstitute.github.io/picard/index.html#GettingHelp";
     }
+
+    /**
+     * Check if R is installed
+     * @param has_chart_output
+     */
+    public static void checkRInstallation(final boolean has_chart_output) {
+        if(has_chart_output) {
+            try {
+                RExecutor.executeFromClasspath("picard/analysis/checkRInstallation.R");
+            } catch (htsjdk.samtools.SAMException e) {
+                throw new PicardException("R is not installed on this machine. It is required for creating the chart.");
```

Comment:
Rather than throwing an exception here, could you return the error string (or just a true / false) from this class if R is not installed?
And then in the calling program's CustomCommandLineValidator code you could just check if checkRInstallation failed and return a string array like:

`         if (!checkRInstallation(CHART_OUTPUT != null)) {
             return new String[]{"R is not installed on this machine. It is required for creating the chart."};
         }`

This is more in line with how CustomCommandLine validation should work.  
I think the usage in CollectInsertSizeMetrics.java is more correct than that in CollectRnaSeqMetrics.java